src/util/namespace.cpp

```diff
@@ -92,6 +92,15 @@ namespace_baset::follow_tag(const c_enum_tag_typet &src) const
   return to_c_enum_type(symbol.type);
 }
 
+const struct_union_typet &follow_struct_union_tag(const tag_typet &src) const
```

Comment:
How about introducing an intermediate `struct_or_union_tag_typet`, which becomes the base case of `struct_tag_typet` and `union_tag_typet`.

Reply:
Indeed, I should have done this right away. Is now included.